src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CallbackValidationsTest.java

```diff
@@ -258,6 +261,79 @@ void noCaseIdTest() {
         );
     }
 
+    @Test
+    void caseIdTest() {
+        checkValidation(
+            createCaseWith(b -> b.id(1L)),
+            true,
+            1L,
+            CallbackValidations::hasAnId,
+            null
+        );
+    }
+
+    @Test
+    void caseIdNoCaseDetailsTest() {
+        checkValidation(
+            null,
+            false,
+            1L,
+            CallbackValidations::hasAnId,
+            "Exception case has no Id"
+        );
+    }
+
+    @Test
+    void hasInvalidDataTest() {
+        CaseDetails caseDetails = createCaseWith(
+            b -> b
+                .id(null)
+                .caseTypeId("SERVICE_ExceptionRecord")
+                .jurisdiction("BULKSCAN")
+        );
+        Validation<Seq<String>, CaseDetails> res =
+            CallbackValidations.hasValidData(true, caseDetails);
+        assertThat(res.isValid()).isEqualTo(false);
+    }
+
+    @Test
+    void hasValidData_use_search_case_reference_true() {
+        Map<String, Object> caseData = new HashMap<>();
+        caseData.put("searchCaseReferenceType", "ccdCaseReference");
+        caseData.put("searchCaseReference", "12345");
+        caseData.put("scannedDocuments", document("https://url", "fileName.pdf"));
+
+        CaseDetails caseDetails = createCaseWith(
+            b -> b
+                .id(1L)
+                .caseTypeId("SERVICE_ExceptionRecord")
+                .jurisdiction("BULKSCAN")
+                .data(caseData)
+        );
+        Validation<Seq<String>, CaseDetails> res =
+            CallbackValidations.hasValidData(true, caseDetails);
+        assertThat(res.isValid()).isEqualTo(true);
+    }
+
+    @Test
+    void hasValidData_use_search_case_reference_false() {
+        Map<String, Object> caseData = new HashMap<>();
+        caseData.put("searchCaseReferenceType", "ccdCaseReference");
+        caseData.put("searchCaseReference", "12345");
+        caseData.put("scannedDocuments", document("https://url", "fileName.pdf"));
+
+        CaseDetails caseDetails = createCaseWith(
+            b -> b
+                .id(1L)
+                .caseTypeId("SERVICE_ExceptionRecord")
+                .jurisdiction("BULKSCAN")
+                .data(caseData)
+        );
+        Validation<Seq<String>, CaseDetails> res =
+            CallbackValidations.hasValidData(true, caseDetails);
```

Comment:
If this was changed to false, I guess the test would fail.

Reply:
Fixed